### DISCUSSION TITLE
`keep_trailing_newline: false` and `include`

### DISCUSSION THREAD

**Discussion by chadxzs** [copier-org/copier#1239] [Q&A]
Hello. I like `keep_trailing_newline: true` (the default in `_envops`) but my editor always adds a trailing newline to all my files and it breaks places where I'm using `{% include 'foo.jinja' %}` because the newline is rendered. I have searched for a way to suppress this newline only for includes but have not found it. Any ideas?

**Comment by pawamoy** [comment]
Tell your editor to stop adding trailing new lines to Jinja files :slightly_smiling_face:

**Reply by chadxzs** [reply]
It's a common configuration, even if I set mine to stop, coworkers will have this happen inevitably.

**Reply by pawamoy** [reply]
Pesky editors :smile: Yeah that's a good point.

**Comment by chadxzs** [comment]
I have worked around the issue for now by switching from including a template to importing a macro. This let me keep the newline configuration as-is.

Fwiw, I used to use when:false on prompt variables with a jinja template as the default. This was much nicer, as I could simply use variable syntax throughout. Disappointed this behavior was changed.

**Comment by yajo** [comment]
Add a - at the end, to remove trailing whitespace: 

```diff
-{% include 'foo.jinja' %}
+{% include 'foo.jinja' -%}
